include/linux/hypercall.h

```diff
@@ -31,7 +31,7 @@ static inline void igloo_hypercall(uint32_t num, uint32_t arg1) {
 #endif
 }
 
-static inline unsigned long igloo_hypercall2(unsigned long num, unsigned long arg1, unsigned long arg2) {
+static inline unsigned long igloo_hypercall2(uint32_t num, uint32_t arg1, uint32_t arg2) {
```

Comment:
Should these be `uint32_t`s? I feel like what we actually want is like TARGET_PTR_t which isn't a linux type, but if we want to support both 32 or 64 bit guests I'd think a 64-bit size would be better?

Reply:
updated!